### PR TITLE
Updates to support Wagtail 5

### PR DESCRIPTION
**Pull request by nickmoreton**
This PR includes: https://github.com/torchbox/wagtail-footnotes/pull/48

Makes changes to:
- Drop support for Django v4.0 and python 3.7

Adds testing to the test matrix for Wagtail 5.0 and Django 4.2

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+repos:
+  - repo: https://github.com/pycqa/isort
+    rev: 5.12.0
+    hooks:
+      - id: isort
+  - repo: https://github.com/psf/black
```
Comment (zerolab) [inline .pre-commit-config.yaml:6]: it is worth revising the verions here

File: tests/settings.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    "wagtail.users",
+    "wagtail.snippets",
+    "wagtail.documents",
+    "wagtail.images",
+    "wagtail.search",
+    "wagtail.admin",
+    "wagtail.api.v2",
+    "wagtail.contrib.modeladmin",
+    "wagtail.contrib.routable_page",
+    "wagtail.contrib.styleguide",
```
Comment (zerolab) [inline tests/settings.py:47]: question: why include the api, routable_page, styleguide, redirects, search_promotions, embeds, Django sitemaps, forms? if the app only looks at a custom StreamField (with rich text)

**Comment by nickmoreton** [comment]
@zerolab This should be good to go now, what do you think please?

**Review by zerolab — APPROVED** [review]
Thank you for this, @nickmoreton.
Approving. I have a few nitpicks but will sort them out

**Pull request merged**